client/src/App.tsx

```diff
@@ -70,64 +84,46 @@ const loadAssetsAsync = async (): Promise<void> => {
 };
 
 function App(): ReactElement {
-  const environment = useRelayEnvironment();
-
-  const [loading, setLoading] = useState<boolean>(false);
   const [assetLoaded, setAssetLoaded] = useState<boolean>(false);
   const { setDeviceType } = useDeviceContext();
   const { setUser } = useAuthContext();
 
-  const responseListener = useRef<any>();
+  const [commitNotification, notificationInFlight] = useMutation<
+    AppCreateNotificationMutation
+  >(createNotificationMutation);
 
   const setDevice = async (): Promise<void> => {
     const deviceType = await Device.getDeviceTypeAsync();
 
     setDeviceType(deviceType);
   };
 
-  const initUser = async (me: AppUserQueryResponse['me']): Promise<void> => {
-    if (!me) return;
-
-    setUser(me as User);
-    setLoading(false);
-  };
+  const { me } = useLazyLoadQuery<AppUserQuery>(meQuery, {});
 
   useEffect(() => {
-    fetchQuery<AppUserQuery>(environment, meQuery, {}).subscribe({
-      start: () => {
-        setLoading(true);
-      },
-      error: (error: Error) => {
-        console.log('error', error);
-        setLoading(false);
-      },
-      next: (data) => {
-        if (data.me) {
-          initUser(data.me);
-
-          return;
-        }
-
-        AsyncStorage.removeItem('token');
-        setDevice();
-      },
-    });
+    if (me === null) {
+      AsyncStorage.removeItem('token');
+      setDevice();
+
+      return;
+    }
+
+    // Update auth context.
+    setUser(me);
 
+    // Register notification token.
     registerForPushNotificationsAsync().then((pushToken) => {
-      if (pushToken) { AsyncStorage.setItem('push_token', pushToken); }
-    });
+      if (pushToken) {
+        AsyncStorage.setItem('push_token', pushToken);
 
-    responseListener.current = Notifications.addNotificationResponseReceivedListener((response) => {
-      console.log(response);
+        commitNotification({
+          variables: { token: pushToken },
+        });
```

Comment:
I think this isn't good idea to save `pushToken` on the server when the user is not signed it.
When you received the `pushToken`, you can save this locally and save this in the server when user signing. Then remove this when the user signout.